assets/js/components/surveys/CurrentSurvey.test.js

```diff
@@ -80,6 +80,167 @@ describe( 'CurrentSurvey', () => {
+			expect(
+				getByLabelText( `Text input for option Other` )
+			).toHaveAttribute( 'disabled' );
+			fireEvent.click( getByText( 'Other' ) );
+			expect(
+				getByLabelText( `Text input for option Other` )
+			).not.toHaveAttribute( 'disabled' );
+
+			// Toggle between Other and other options
+			fireEvent.click( getByText( 'Satisfied' ) );
+			expect(
+				getByLabelText( `Text input for option Other` )
+			).toHaveAttribute( 'disabled' );
+			fireEvent.click( getByText( 'Other' ) );
+			expect(
+				getByLabelText( `Text input for option Other` )
+			).not.toHaveAttribute( 'disabled' );
+			fireEvent.click( getByText( 'Neutral' ) );
+			expect(
+				getByLabelText( `Text input for option Other` )
+			).toHaveAttribute( 'disabled' );
+			fireEvent.click( getByText( 'Other' ) );
+			expect(
+				getByLabelText( `Text input for option Other` )
+			).not.toHaveAttribute( 'disabled' );
+
+			// Submit button should now be disabled until text is entered
+			expect( getByRole( 'button', { name: 'Next' } ) ).toHaveAttribute(
+				'disabled'
+			);
+
+			fireEvent.change( getByLabelText( `Text input for option Other` ), {
+				target: { value: 'foo' },
+			} );
+
+			expect(
+				getByRole( 'button', { name: 'Next' } )
+			).not.toHaveAttribute( 'disabled' );
+
+			fireEvent.change( getByLabelText( `Text input for option Other` ), {
+				target: { value: STRING_110_CHARACTERS },
+			} );
+
+			expect(
+				getByLabelText( `Text input for option Other` )
+			).toHaveValue( STRING_100_CHARACTERS );
+		} );
+
+		it( 'should submit answer in correct shape', async () => {
+			const { getByText, getByRole, getByLabelText, findByText } = render(
+				<CurrentSurvey />,
+				{
+					registry,
+				}
+			);
+
+			expect( fetchMock ).toHaveBeenCalledTimes( 1 );
+
+			fireEvent.click( getByText( 'Other' ) );
+
+			fireEvent.change( getByLabelText( `Text input for option Other` ), {
+				target: { value: STRING_110_CHARACTERS },
+			} );
+
+			// Check that submits correctly
+			fireEvent.click( getByRole( 'button', { name: 'Next' } ) );
+
+			expect( fetchMock ).toHaveBeenCalledTimes( 2 );
+
+			expect( fetchMock ).toHaveFetched(
+				'/google-site-kit/v1/core/user/data/survey-event?_locale=user',
+				{
+					credentials: 'include',
+					method: 'POST',
+					body: {
+						data: {
+							event: {
+								question_answered: {
+									question_ordinal: 1,
+									answer: {
+										answer: {
+											answer_ordinal: 6,
+											answer_text:
+												'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec suscipit auctor dui, id faucibus nisl',
+										},
+									},
+								},
+							},
+							session: {
+								session_id: 'storybook_session',
+								session_token: 'token_12345',
+							},
+						},
+					},
+					headers: {
+						Accept: 'application/json, */*;q=0.1',
+						'Content-Type': 'application/json',
+					},
+				}
+			);
+
+			await findByText( 'Thanks for sharing your thoughts!' );
```

Comment:
We should put an `expect` call around this, something like:

```suggestion
			const completionMessage = await findByText( 'Thanks for sharing your thoughts!' );
			expect( completionMessage ).not.toBeEmpty();
```

Or something. Right now this operates as an implicit test but let's make it explicit.